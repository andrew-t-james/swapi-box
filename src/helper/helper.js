export const fetchHelper = async url => {
  const getData = await fetch(url);
  const response = await getData.json();
  return response;
};

export const getPersonData = async personData => {
  const personSpecies = personData.map(async people => {
    const getPersonSpecies = await fetchHelper(people.species[0]);
    const getHomeWorld = await fetchHelper(people.homeworld);
    return {
      name: people.name,
      species: getPersonSpecies.name,
      homeworld: getHomeWorld.name,
      population: getHomeWorld.population
    };
  });
  return Promise.all(personSpecies);
};