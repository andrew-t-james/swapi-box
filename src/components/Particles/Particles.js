import React from 'react';
import Particles from 'react-particles-js';
import { particlesConfig } from '../../helper/particlesjs-config';

const ParticleBackground = () => {
  return (
    <Particles params={particlesConfig} />
  );
};

export default ParticleBackground;
