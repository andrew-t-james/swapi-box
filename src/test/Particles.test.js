import React from 'react';
import { shallow } from 'enzyme';
import Particles from '../components/Particles/Particles';
import { particlesConfig } from '../helper/particlesjs-config';

describe('Particles', () => {
  let wrapper;


  beforeEach(() => wrapper = shallow(
    <Particles
      params={particlesConfig}
    />
  ));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
