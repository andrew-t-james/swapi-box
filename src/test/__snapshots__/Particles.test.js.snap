// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Particles should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ParticleBackground
    params={
        Object {
            "interactivity": Object {
              "detect_on": "canvas",
              "events": Object {
                "onclick": Object {
                  "enable": false,
                  "mode": "push",
                },
                "onhover": Object {
                  "enable": false,
                  "mode": "repulse",
                },
                "resize": true,
              },
              "modes": Object {
                "bubble": Object {
                  "distance": 400,
                  "duration": 2,
                  "opacity": 8,
                  "size": 40,
                  "speed": 3,
                },
                "grab": Object {
                  "distance": 400,
                  "line_linked": Object {
                    "opacity": 1,
                  },
                },
                "push": Object {
                  "particles_nb": 4,
                },
                "remove": Object {
                  "particles_nb": 2,
                },
                "repulse": Object {
                  "distance": 200,
                  "duration": 0.4,
                },
              },
            },
            "particles": Object {
              "color": Object {
                "value": "#ffffff",
              },
              "line_linked": Object {
                "color": "#ffffff",
                "distance": 150,
                "enable": false,
                "opacity": 0.4,
                "width": 1,
              },
              "move": Object {
                "attract": Object {
                  "enable": false,
                  "rotateX": 600,
                  "rotateY": 868.0624057955,
                },
                "bounce": false,
                "direction": "none",
                "enable": true,
                "out_mode": "out",
                "random": false,
                "speed": 3.1565905665290903,
                "straight": false,
              },
              "number": Object {
                "density": Object {
                  "enable": true,
                  "value_area": 800,
                },
                "value": 80,
              },
              "opacity": Object {
                "anim": Object {
                  "enable": false,
                  "opacity_min": 0.1,
                  "speed": 1,
                  "sync": false,
                },
                "random": false,
                "value": 0.5,
              },
              "shape": Object {
                "image": Object {
                  "height": 100,
                  "src": "img/github.svg",
                  "width": 100,
                },
                "polygon": Object {
                  "nb_sides": 5,
                },
                "stroke": Object {
                  "color": "#000000",
                  "width": 0,
                },
                "type": "circle",
              },
              "size": Object {
                "anim": Object {
                  "enable": false,
                  "size_min": 0.1,
                  "speed": 40,
                  "sync": false,
                },
                "random": true,
                "value": 3,
              },
            },
            "retina_detect": true,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "particles",
      "height": "100%",
      "params": Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": false,
              "mode": "push",
            },
            "onhover": Object {
              "enable": false,
              "mode": "repulse",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 400,
              "duration": 2,
              "opacity": 8,
              "size": 40,
              "speed": 3,
            },
            "grab": Object {
              "distance": 400,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 200,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 150,
            "enable": false,
            "opacity": 0.4,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 868.0624057955,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "out",
            "random": false,
            "speed": 3.1565905665290903,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 800,
            },
            "value": 80,
          },
          "opacity": Object {
            "anim": Object {
              "enable": false,
              "opacity_min": 0.1,
              "speed": 1,
              "sync": false,
            },
            "random": false,
            "value": 0.5,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.svg",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#000000",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": false,
              "size_min": 0.1,
              "speed": 40,
              "sync": false,
            },
            "random": true,
            "value": 3,
          },
        },
        "retina_detect": true,
      },
      "style": Object {},
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "particles",
        "height": "100%",
        "params": Object {
          "interactivity": Object {
            "detect_on": "canvas",
            "events": Object {
              "onclick": Object {
                "enable": false,
                "mode": "push",
              },
              "onhover": Object {
                "enable": false,
                "mode": "repulse",
              },
              "resize": true,
            },
            "modes": Object {
              "bubble": Object {
                "distance": 400,
                "duration": 2,
                "opacity": 8,
                "size": 40,
                "speed": 3,
              },
              "grab": Object {
                "distance": 400,
                "line_linked": Object {
                  "opacity": 1,
                },
              },
              "push": Object {
                "particles_nb": 4,
              },
              "remove": Object {
                "particles_nb": 2,
              },
              "repulse": Object {
                "distance": 200,
                "duration": 0.4,
              },
            },
          },
          "particles": Object {
            "color": Object {
              "value": "#ffffff",
            },
            "line_linked": Object {
              "color": "#ffffff",
              "distance": 150,
              "enable": false,
              "opacity": 0.4,
              "width": 1,
            },
            "move": Object {
              "attract": Object {
                "enable": false,
                "rotateX": 600,
                "rotateY": 868.0624057955,
              },
              "bounce": false,
              "direction": "none",
              "enable": true,
              "out_mode": "out",
              "random": false,
              "speed": 3.1565905665290903,
              "straight": false,
            },
            "number": Object {
              "density": Object {
                "enable": true,
                "value_area": 800,
              },
              "value": 80,
            },
            "opacity": Object {
              "anim": Object {
                "enable": false,
                "opacity_min": 0.1,
                "speed": 1,
                "sync": false,
              },
              "random": false,
              "value": 0.5,
            },
            "shape": Object {
              "image": Object {
                "height": 100,
                "src": "img/github.svg",
                "width": 100,
              },
              "polygon": Object {
                "nb_sides": 5,
              },
              "stroke": Object {
                "color": "#000000",
                "width": 0,
              },
              "type": "circle",
            },
            "size": Object {
              "anim": Object {
                "enable": false,
                "size_min": 0.1,
                "speed": 40,
                "sync": false,
              },
              "random": true,
              "value": 3,
            },
          },
          "retina_detect": true,
        },
        "style": Object {},
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
